blinkpy/sync_module.py

```diff
@@ -209,17 +268,29 @@ def refresh(self, force_cache=False):
 
     def check_new_videos(self):
         """Check if new videos since last refresh."""
+        _LOGGER.debug("Checking for new videos")
         try:
             interval = self.blink.last_refresh - self.motion_interval * 60
+            last_refresh = datetime.datetime.fromtimestamp(self.blink.last_refresh)
+            _LOGGER.debug(f"last_refresh = {last_refresh}")
+            _LOGGER.debug(f"interval={interval}")
         except TypeError:
             # This is the first start, so refresh hasn't happened yet.
             # No need to check for motion.
+            e = traceback.format_exc()
+            _LOGGER.error(
+                f"Error calculating interval (last_refresh={self.blink.last_refresh}): {e}"
+            )
+            trace = "".join(traceback.format_stack())
+            _LOGGER.debug(f"\n{trace}")
+            _LOGGER.info("No new videos since last refresh.")
             return False
 
         resp = api.request_videos(self.blink, time=interval, page=1)
 
         for camera in self.cameras.keys():
             self.motion[camera] = False
+            self.last_records[camera] = []
```

Comment:
This is the line that resets the last records when refresh is called.  Under what conditions should I _not_ reset the last records list?